core/controllers/base_test.py

```diff
@@ -1423,3 +1469,407 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
         with self.swap(self, 'testapp', self.mock_testapp):
             self.post_json(
                 '/correctmock', payload, expected_status_int=200)
+
+
+class SchemaValidationIntegrationTests(test_utils.GenericTestBase):
+    """Tests all the functionality of SVS(Schema-Validation-System)
+    architecture.
+    """
+
+    non_schema_handlers = payload_validator.NON_SCHEMA_HANDLERS
+    non_schema_requiring_handlers = (
+            payload_validator.NON_SCHEMA_REQUIRING_HANDLERS)
+    handlers_need_schema = []
+    handlers_to_remove = []
+    handlers_with_missing_url_schema_keys = []
+    handlers_with_missing_req_schema_keys = []
+    handlers_with_non_conforming_default_schema = []
+
+    wiki_page_link = (
+        'https://github.com/oppia/oppia/wiki/Validation-of-handler-args')
+
+    def _every_handler_class_has_schema(self, handler, handler_class_name):
+        """This test ensures that every child class of BaseHandler must have
+        schema defined into it.
+
+        Args:
+            handler: BaseHandler. A callable to handle the route.
+            handler_class_name: str. Name of the handler class.
+        """
+        if handler_class_name in self.non_schema_handlers:
+            return
+
+        schema_written_for_request_methods = (
+            handler.HANDLER_ARGS_SCHEMAS !=
+                base.BaseHandler.HANDLER_ARGS_SCHEMAS)
```

Comment:
Just use None here and below for the compared-to value. Ultimately what you're trying to check for is just that the schema isn't None, and it saves the reader having to go and look up the base value (which could potentially change in the future).

Reply:
Done